libraries/botbuilder/src/cloudAdapter.ts

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/**
+ * @module botbuilder
+ */
+
+import { BotFrameworkAuthentication } from 'botframework-connector';
+import { BotFrameworkHttpAdapter } from './botFrameworkHttpAdapter';
+import { CloudAdapterBase } from './cloudAdapterBase';
+import { StatusCodes } from 'botbuilder-core';
+import { readRequest, writeResponse } from './httpHelper';
+import { tests } from 'botbuilder-stdlib';
+
+import {
+    BotLogic,
+    Emitter,
+    Request,
+    Response,
+    assertBotLogic,
+    assertEmitter,
+    assertRequest,
+    assertResponse,
+} from './interfaces';
+
+export class CloudAdapter extends CloudAdapterBase implements BotFrameworkHttpAdapter {
+    /**
+     * Construct a CloudAdapter.
+     *
+     * @param {BotFrameworkAuthentication} botFrameworkAuthentication the bot framework authentication, optional
+     */
+    constructor(botFrameworkAuthentication: BotFrameworkAuthentication);
+    constructor(botFrameworkAuthentication: unknown) {
```

Comment:
This pattern, which shows up throughout this PR, forces us to use type assertions for proper runtime validation. This is useful because we support non-Typescript users and already do some runtime validation. Using `unknown` and type assertions is a more formal way of enforcing this contract.